File: src/c++/dynamic_programming/11047/main.cc
/* Date : 2019.5.18
 * Author : seungyeon choi
 * 11047 (동전0)
 */

#include <cstdio>

using namespace std;

const int MAX_COUNT = 11;

int N, K, ans;
int COIN[MAX_COUNT];

int main() {
  scanf("%d %d", &N, &K);

  for (int i = 0; i < N; i++)
    scanf("%d", &COIN[i + 1]);

  while (K) {
    for (int i = N; i > 0; i--) {
      if (COIN[i] <= K) {
        ans += (K / COIN[i]);
        K = (K % COIN[i]);
      }
    }
  }

  printf("%d\n", ans);

  return 0;
}

File: src/python/dynamic_programming/11047/main.py
# Date : 2017.11.5
# Author : seungyeon choi
# 11047 (동전0)

temp = list(map(int, input().split()))
N, V = temp[0], temp[1]
coin = [0 for i in range(N+1)]
for i in range(1, N+1):
    coin[i] = int(input())
ans = 0
while V :
    for i in range(N, 0, -1):
        if coin[i] <= V and N == i:
            ans += int(V / coin[i])
            V = (V % coin[i])
        if coin[i] <= V :
            ans += int(V / coin[i])
            V = (V % coin[i])
print(ans)
